app/src/main/java/io/apicurio/registry/mt/TenantIdResolver.java

```diff
@@ -74,6 +85,7 @@ public boolean resolveTenantId(String uri, Supplier<String> tenantIdHeaderProvid
                 // 1 is t
                 // 2 is the tenantId
                 String tenantId = tokens[TENANT_ID_POSITION];
+                checkTenantAuthorization(tenantId);
```

Comment:
this check is also required below. I suggest moving this code into `TenantContextLoader#loadContext` this way regardless of where the context is loaded the auth check is performed, that will also serve as an optimization because we will only make one http request to tenant-manager to get the tenant info and then perform the auth check and load the context with the same data.